packages/account/src/providers/utils/extract-tx-error.ts

```diff
@@ -56,19 +56,23 @@ export const assembleRevertError = (
 
   if (revertReceipt) {
     const reasonHex = bn(revertReceipt.val).toHex();
+    const lastLog = logs[logs.length - 1];
+    const lastButOneLog = logs[logs.length - 2];
 
     switch (reasonHex) {
       case FAILED_REQUIRE_SIGNAL: {
         reason = 'require';
         errorMessage = `The transaction reverted because a "require" statement has thrown ${
-          logs.length ? stringify(logs[0]) : 'an error.'
+          logs.length ? stringify(lastLog) : 'an error.'
```

Comment:
@maschad When Sway `require` throws an error, it logs the value passed as its second parameter.

A similar thing happens with both `assert_eq` and `assert_ne`, which will log 2 values in these cases.

The logged value(s) will be at the end of the logs, as it will be the last thing logged before the transaction reverts.

That's why I have added [multiple logs](https://github.com/FuelLabs/fuels-ts/blob/acb3ad2d0a0a3f46aaf26a279e452b606e66a951/packages/fuel-gauge/test/fixtures/forc-projects/revert-error/src/main.sw) to the contracts functions, to ensure that it will always assemble the error with the last log entries.

Reply:
Thanks for the clarification.